default/scripting/species/common/population.macros

```diff
@@ -129,7 +129,7 @@ SELF_SUSTAINING_BONUS
                 //TargetPopulation low = 0
             ]
             accountinglabel = "SELF_SUSTAINING_LABEL"
-            priority = [[VERY_LATE_PRIORITY]]
+            priority = [[EARLY_PRIORITY]]
```

Comment:
To be consistent with the idea (noted in the script comment) that self-sustaining is meant to be equivalent to having 3 growth specials, then if you change this then you'd also need to change growth specials to have them apply early also, or you wind up nerfing the self-sustaining max pop relative to the other metabolisms.  And I think we'd have to look really closely at things if we considered changing the timing of the growth specials.  

A lot of time went into a previous pass at getting the various growth techs & specials to work out right; since it seeems you are really just looking for another way to nerf the Sly here, I think it would probably be better to look a little farther and leave self-sustaining alone.

Reply:
Ah because nobody objected to the idea i thought i take it in. I take the commit out then.

Reply:
Well, let's see about getting some other feedback-- as I think about it more, this would not make self-sustaining *always* worse than the other metabolisms, in the case of a Good Pop species this would give self-sustaining a bigger boost than other metabolisms.  So perhaps we just change the code comment to say self-sustaining is meant to be "roughly equivalent" to having three growth specials. 
@Vezzra @MatGB I recall you guys putting a fair bit of thought into Population scripting, do you guys have an opinion on this?

Reply:
Regardless of whether or not the change to self-sustaining priority makes sense in general, I really don't think it should be a part of this PR.

Please also note that effect timing may (and in this case does) need tweaks in the AI code.

Reply:
> @Vezzra @MatGB I recall you guys putting a fair bit of thought into Population scripting, do you guys have an opinion on this?

Actually that has been @MatGB (I haven't fiddled with the pop scripting), so I guess he might want to say something on the subject.

But of course I have an opinion on this as well, I agree with you, @Dilvish-fo, and @Morlic-fo: we should leave the self-sustaining metabolism scripts alone. If the Sly need some nerfing wrt to their pop levels, I suggest to just do that by giving them the bad or very bad pop trait.

If that isn't sufficient, add bad industry/research. There are plenty of species traits we can use to fine-tune them. But please refrain from changing things that will also change other things (by that I also refer to the approach of changing the planet size of gas giants!).

Trying to make a species balanced by fiddling around with settings/scripting that affects/changes other species or game elements as well carries a great potential of introducing a lot of issues. Which, IMO, should be avoided.

Reply:
Regarding changing the gas giants size - as @Dilvish-fo is on board for the AI team this is Ok, right? 

@Vezzra probably you dont have much time at your hands right now for the project. I and others spend a lot of time on discussions, and then there is a conclusion/decision. And then work gets done to have it implemented. If you rightly object in this stage, a lot of effort is wasted. If you can chime in earlier it would be greatly appreciated.

Reply:
>  (by that I also refer to the approach of changing the planet size of gas giants!)

As I understand it, the planet size is not changed, only the value returned from [`SizeAsInt`](https://github.com/freeorion/freeorion/blob/master/universe/Planet.cpp#L203), which is not the same as [`PlanetSize`](https://github.com/freeorion/freeorion/blob/master/universe/Enums.h#L59) cast to int.

Propose `SizeAsInt` function/keywords should be renamed to avoid confusion (HabitableSize?), such need not be done in this PR imo.

Reply:
@agrrr3, I understand the frustration, but as everyone else I'm doing this in my free time for fun, and life tends to get in the way much more than I'd like. I'm trying, but there is only so much priority I'm willing to give something that's just a hobby, after all.

Which is why I don't expect anyone to receive my input as something they must abide, but as mere suggestions/proposals. 😉 

That said, in this particular case I responded to a request to comment that had only been made a day ago, so that couldn't have been too much of a problem - unless you refer to my objection to the change of the planet size of gas giants. In this case I'm sorry, I didn't realize that reverting that decision (and trying a different approach) would undo so much work.

It certainly can remain, at least for the time being, as a stop-gap solution until we can come up with better ideas how to balance the Sly. However, regardless if I mentioned it or not, the issue I pointed out is still true, and sooner or later we'd have to put in the work to solve this properly anyway. It's not my fault no one else noticed the problem with that approach... 😉

Reply:
@dbenage-cx:

> As I understand it, the planet size is not changed, only the value returned from [`SizeAsInt`](https://github.com/freeorion/freeorion/blob/master/universe/Planet.cpp#L203), which is not the same as [`PlanetSize`](https://github.com/freeorion/freeorion/blob/master/universe/Enums.h#L59) cast to int.

Well, but shouldn't it be? I always thought `SizeAsInt` has been added to provide content scripts with the means to use the planet size in formulas. Having `SizeAsInt` returning the same value for gas giants and large planets is asking for (future) trouble IMO. Right now the planet size of gas giants isn't used anywhere else (IIRC), so it won't cause any immediate side effects, but that can very well change in the future.

> Propose `SizeAsInt` function/keywords should be renamed to avoid confusion (HabitableSize?)

I wouldn't change `SizeAsInt`, but *add* a function/keyword `HabitableSize`, so you can still refer to the actual planet size in a formula, as well as the "population capacity". That should go a long way to mitigate a lot of potential issues.

Nevertheless, even with that distinction available, I'm still hesitant to make gas giants like large planets wrt to pop capacity just because we have trouble balancing one species. Not a big issue though, and certainly not something we need to address right now. Plenty of time for that should there ever arise any problems.

> such need not be done in this PR imo

Agreed. But it definitely should be done at some point, I'd even suggest to open an issue for that (otherwise it will surely be forgotten 😉).